spec/requests/controllers_requests_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+RSpec.describe 'Combined Specs', type: :request do
+  # Users Specs
+  context 'GET /users' do
+    it 'renders a list of users' do
+      # Create some test users directly in the database
+      User.create(name: 'kamo')
+      User.create(name: 'kagiso')
+      User.create(name: 'kayjeee')
+
+      # Make a GET request to the users index page
+      get users_path
+
+      # Assert that the response is successful (HTTP status 200)
+      expect(response).to have_http_status(:ok)
+
+      # Assert that the correct template is rendered
+      expect(response).to render_template('users/index')
+
+      # Assert that the page contains the names of the test users
+      expect(response.body).to include('otmane')
+      expect(response.body).to include('Yusuf')
+      expect(response.body).to include('Rayan')
+    end
+  end
```

Comment:
Nice job writing some test cases for your user controllers👍🏼. There are one vital suggestions that you can kindly attend to😉.

some of your test cases for your user controller are failing, Kindly do well to ensure all test cases are passing before proceeding to submit👍🏼.
<img width="832" alt="image" src="https://github.com/kayjeee/Hi-Blog_app_kayjeee/assets/64643995/c4472dce-1eb9-4d36-ae9b-a9c4563a6aa2">